### DISCUSSION TITLE
Browser navigation in between app directory pages are cached and I cannot opt out

### DISCUSSION THREAD

**Discussion by priosshrsth** [vercel/next.js#70422] [Help]
### Link to the code that reproduces this issue

https://github.com/priosshrsth/next-cache-test

### To Reproduce

1. Any next js app with app directory support.( I Tried with 14 & 15)
2. Create multiple app routes `/` & `/contact`
3. Navigating to /contact from home page
4. Click back button on the browser. 

### Current vs. Expected behavior

The homepage after browser navigation is not latest and is cached page. No code in` /page.tsx` runs and no fetch requests are made.

Expected behavior: Same thing with pages route and `getServerSideProps` shows updated data. And If I put `console.log `inside `getServerSideProps`, it runs. Also, navigation back from pages route to homepage also returns updated data. This is so bad and so wierd. I don't want to migrate my whole app to pages directory now.

### Provide environment information

```bash
Mac OS
Brave browser
```

### Which area(s) are affected? (Select all that apply)

Navigation

### Which stage(s) are affected? (Select all that apply)

next dev (local), next build (local), next start (local), Vercel (Deployed)

### Additional context

I tested on next js 14 and 15. I have used everything like export const dynamic = 'forcedynamic', settint revalidate and everything. Nothing works. meta tag anc cache headers are also set to not cache as well.

**Comment by tang-fe** [comment]
This looks like the browser's bfcache https://web.dev/articles/bfcache

**Comment by priosshrsth** [comment]
> https://web.dev/articles/bfcache

You might be correct. How can I verify this. Adding pageshow event listener doesn't work. As this does not get retriggered on navigation. Also, I tried adding `beforeunload`, but it doesn't seem to disable bfcache. And also it works when navigating from pages routes to app routes in brave browser. But on safari it seels, they are always cached on browser navigation.

**Comment by priosshrsth** [comment]
```
import { type PropsWithChildren, useEffect, useState } from "react";
import { usePathname, useRouter } from "next/navigation";

export function BackButtonCacheTest({ children }: PropsWithChildren) {
  const router = useRouter();
  const pathname = usePathname();
  const [isBrowserNav, setIsBrowserNav] = useState(false);

  useEffect(() => {
    window.addEventListener("popstate", () => {
      setIsBrowserNav(true);
    });
  }, [router]);

  useEffect(() => {
    if (!isBrowserNav) {
      return;
    }

    setTimeout(() => {
      setIsBrowserNav(false);
      router.refresh();
    }, 500);
  }, [pathname, router, isBrowserNav]);

  return (
    <>
      {isBrowserNav && (
        <div className="pointer-events-auto fixed inset-0 z-50 h-screen w-screen bg-black bg-opacity-50" />
      )}
      {children}
    </>
  );
}
```

Wrapping children in layout in this component seem to work. Maybe I can work on it to improve and make it more predictable.

**Comment by samcx** [comment]
@priosshrsth I'm pretty sure expected caching behavior is happening here, but can you simplify your :repro: so we can confirm?

**Reply by priosshrsth** [reply]
@samcx Yes. I will try to find some time today to create a simple minimal repo.
